### PR TITLE
18 ajouter un code led pour chaque action

### PR DESCRIPTION
**Pull request by MarsiaSec**

### CODE REVIEW

File: arduino/main/src/Leds.h [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+        
+        /**
+         * Initializing sequence that plays while the robot is waiting for the RPi to initialize.
+         */
+        void initializing(int interval, unsigned long time);
+
+        /**
+         * Given the color name as a string, returns it's `uint32_t` representation.
+         * This method is used so that a simple string can be provided as a parameter
+         *
```
Comment (lbo462) [inline arduino/main/src/Leds.h:54]: You can remove this line

File: arduino/main/src/RescueBot.h [hunk 2]
```diff
@@ -70,6 +75,11 @@ class RescueBot
          */
         void explore();
 
+        /**
+         * Controls the leds on top of the robot.
+         */
+        //void leds();
```
Comment (lbo462) [inline arduino/main/src/RescueBot.h:81]: If you do not use this, you can remove it

File: docs/DevEnv.md [hunk 3]
```diff
@@ -17,9 +17,40 @@ Otherwise, check or change its location in the preferences in the Arduino IDE.
...
 You'll find the extension by typing `Arduino` in the VSCode search extension field 
 (`Ctrl+Shift+X`) and clicking the one provided by Microsoft.
 
+Next, copy the `libraries` directory from the repo to `~/Arduino/libraries/` like so:
+
+```bash
+cp -r libraries ~/Arduino/libraries
+```
+
+You will also need to update the `includePath` by adding the path the `c_cpp_properties.json` file:
```
Comment (lbo462) [inline docs/DevEnv.md:30]: You might* also need [...]
On Linux, even with a fresh installation without the Arduino IDE, I didn't need to do that.

Plus, you didn't say where to put this file. Is this a VSCode config ?

File: docs/DevEnv.md [hunk 4]
```diff
@@ -17,9 +17,40 @@ Otherwise, check or change its location in the preferences in the Arduino IDE.
...
+            ],
+            "forcedInclude": [],
+            "cStandard": "c17",
+            "cppStandard": "gnu++17",
+            "defines": []
+        },
+        ...
+```
+
+> **NEED TO CHECK IF IT WORKS CORRECTLY**
```
Comment (lbo462) [inline docs/DevEnv.md:52]: Thus, should we put this on master now ?
Are we sure of this ?

**Review by lbo462 — CHANGES_REQUESTED** [review]

File: arduino/main/src/Leds.cpp [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+    // Gets the `uint32_t` color value based on the color provided as a parameter. 
+    uint32_t colorValue = getColor(color);
+
+    // Checking if the the time provided in the `interval` parameter has passed.
+    // This avoids using delays and thus blocking the code.
+    if (time - previousLedMillis >= interval)
+    {
+        previousLedMillis = time;
+
+        if (ledIsOn == 0)
```
Comment (lbo462) [inline arduino/main/src/Leds.cpp:25]: You can just do `if(ledIsOn)` since it's a boolean

File: arduino/main/src/Leds.cpp [hunk 6]
```diff
@@ -0,0 +1,130 @@
...
+    // Gets the `uint32_t` color value based on the color provided as a parameter.
+    uint32_t colorValue = getColor("magenta");
+
+    // Checking if the the time provided in the `interval` parameter has passed.
+    // This avoids using delays and thus blocking the code.
+    if (time - previousLedMillis >= interval)
+    {
+        previousLedMillis = time;
+
+        if (ledIsOn == 0)
```
Comment (lbo462) [inline arduino/main/src/Leds.cpp:114]: Same here

File: arduino/main/src/Leds.cpp [hunk 7]
```diff
@@ -0,0 +1,130 @@
...
+        {
+            // Lights up LEDs incrementally 
+            for (int i = 0; i < ledStrip.numPixels() + 1; i++)
+            {
+                ledStrip.fill(colorValue, 0, i);
+                ledStrip.show();
+                ledStrip.clear();
+                // TEST:
```
Comment (lbo462) [inline arduino/main/src/Leds.cpp:122]: Test this or remove it

**Review by lbo462 — CHANGES_REQUESTED** [review]

File: arduino/main/src/RescueBot.h [hunk 8]
```diff
@@ -78,6 +82,9 @@ class RescueBot
         /** Tells if the object was found. */
         bool objectFound = false;
 
+        /** Is an object detected or not \'O'/ */
+        bool objectDetected = false;
```
Comment (lbo462) [inline arduino/main/src/RescueBot.h:86]: Changed this with selfFound and otherFound (or something like it) to make it clearer

File: arduino/main/src/RescueBot.h [hunk 9]
```diff
@@ -98,6 +105,13 @@ class RescueBot
 
         /** Configures the camera positions */
         CameraPosition camPosition = CameraPosition();
+        
+        /** Params to control the LEDs */
+        const int ledPin=2, numLeds=33, brightness=50;
```
Comment (lbo462) [inline arduino/main/src/RescueBot.h:110]: Should be here

File: arduino/main/src/RescueBot.cpp [hunk 10]
```diff
@@ -69,12 +73,27 @@ void RescueBot::update()
...
+    {
+        ledStrip.blink("green", 200, currentMillis);
+        return;
+    }
+
+    if (objectFound)
+    {
+        smartMotors.stop();
+        ledStrip.rainbow(500, currentMillis);
+        //delay(10000);
```
Comment (lbo462) [inline arduino/main/src/RescueBot.cpp:92]: Remove this

File: arduino/main/src/RescueBot.cpp [hunk 11]
```diff
@@ -69,12 +73,27 @@ void RescueBot::update()
     
 
     // If the RPI isn't ready or if the object was found, just don't move and exit
-    if(!RPIInitialized || objectFound)
+    if(!RPIInitialized)
```
Comment (lbo462) [inline arduino/main/src/RescueBot.cpp:76]: Why ? You should stop the motors if you found something too

File: arduino/main/src/RescueBot.cpp [hunk 12]
```diff
@@ -7,23 +7,27 @@ RescueBot::~RescueBot() {}
 
 void RescueBot::setup()
 {
+    ledStrip.setup();
+    ledStrip.starflitRedToBlue();
     smartMotors.setup();
     ultrasonicSensors.setup();
     camPosition.setup();
+    ledStrip.setup();
```
Comment (lbo462) [inline arduino/main/src/RescueBot.cpp:15]: Duplicated

File: arduino/main/src/Leds.h [hunk 13]
```diff
@@ -0,0 +1,122 @@
...
+        /**
+         * ledStrip object initialization.
+         */
+        Adafruit_NeoPixel ledStrip = Adafruit_NeoPixel(numLeds, ledPin, NEO_GRB + NEO_KHZ800);
+
+        /**
+         * Key value pairs of all the colors you can use associated with their `uint32_t` representation.
+         * Feel free to add additional colors based on your needs.
+         */
+        Colors colors[4] = {
```
Comment (lbo462) [inline arduino/main/src/Leds.h:113]: Shouldn't this be outside of the class too ?

**Review by lbo462 — CHANGES_REQUESTED** [review]

File: arduino/main/src/RescueBot.h (hunk 9 above)
Comment (lbo462) [reply arduino/main/src/RescueBot.h:110]: Should not*

**Review by MarsiaSec — COMMENTED** [review]
ok

**Review by lbo462 — APPROVED** [review]

**Pull request merged**